src/codegen/function.rs

```diff
@@ -49,6 +51,9 @@ pub fn generate(
     let suffix = if only_declaration { ";" } else { " {" };
 
     try!(writeln!(w, ""));
+    if only_declaration {
```

Comment:
Need be `!in_trait || only_declaration`, or `g_settings_list_relocatable_schemas` will be not deprecated in gio.